src/main/resources/templates/db_scripts/schema.json

```diff
@@ -1,4 +1,16 @@
 {
+  "scripts": [
```

Comment:
Shouldn't we be using RMB method like this https://github.com/folio-org/mod-users/blob/14367dbdd1da9afecca1d112756a0b34187129ea/src/main/resources/templates/db_scripts/schema.json#L111 as mentioned in the story "Approach"?

Reply:
In case of using the script, the view will look like in the first variant, it allows you to make queries both by entry and by acq-units to the "metadata" column.

In the second case, the third column contains only information about acq-units-assignments. In this case, the query to the database becomes more complicated.

1. With script snippet

| id  | jsonb | metadata |
| ------------- | -------------------------- | ------------- |
| a9b99f8a-… | { <br /> &nbsp;&nbsp;"id": "a9b99f8a-…", <br />  &nbsp;&nbsp;"accountingCode": "1234", <br />&nbsp;&nbsp;"amount": 23.45,  <br />&nbsp;&nbsp;"batchNumber": "101",  <br /> &nbsp;&nbsp;"disbursementNumber": "EFT5467",  <br />&nbsp;&nbsp;"disbursementDate": "2019-05-05",   <br />&nbsp;&nbsp;"disbursementAmount": 5,  <br />&nbsp;&nbsp;"exchangeRate": 0.89,  <br />&nbsp;&nbsp;"exportToAccounting": false,   <br />&nbsp;&nbsp;"invoiceCurrency": "USD",   <br />&nbsp;&nbsp;"invoiceId": "733cafd3-…",  <br />&nbsp;&nbsp;"status": "Awaiting payment",  <br />&nbsp;&nbsp;"systemCurrency": "EUR",  <br /> &nbsp;&nbsp;"type": "Payment",  <br />&nbsp;&nbsp;"voucherDate": "2019-05-06",  <br /> &nbsp;&nbsp;"voucherNumber": "1000",   <br />&nbsp;&nbsp;"metadata": {     <br />&nbsp;&nbsp;&nbsp;&nbsp;"createdDate": "2019-05-06     <br />&nbsp;&nbsp;&nbsp;&nbsp;"createdByUserId": "28d1057c-…"  <br />&nbsp;&nbsp;}<br /> }  | { <br /> &nbsp;&nbsp;"id": "a9b99f8a-…", <br />  &nbsp;&nbsp;"accountingCode": "1234", <br />&nbsp;&nbsp;"amount": 23.45,  <br />&nbsp;&nbsp;"batchNumber": "101",  <br /> &nbsp;&nbsp;"disbursementNumber": "EFT5467",  <br />&nbsp;&nbsp;"disbursementDate": "2019-05-05",   <br />&nbsp;&nbsp;"disbursementAmount": 5,  <br />&nbsp;&nbsp;"exchangeRate": 0.89,  <br />&nbsp;&nbsp;"exportToAccounting": false,   <br />&nbsp;&nbsp;"invoiceCurrency": "USD",   <br />&nbsp;&nbsp;"invoiceId": "733cafd3-…",  <br />&nbsp;&nbsp;"status": "Awaiting payment",  <br />&nbsp;&nbsp;"systemCurrency": "EUR",  <br /> &nbsp;&nbsp;"type": "Payment",  <br />&nbsp;&nbsp;"voucherDate": "2019-05-06",  <br /> &nbsp;&nbsp;"voucherNumber": "1000", <br />&nbsp;&nbsp;"recordId": "a9b99f8a-…", <br />&nbsp;&nbsp;"acquisitionsUnitId": "0ebb1f7d-…",   <br />&nbsp;&nbsp;"metadata": {     <br />&nbsp;&nbsp;&nbsp;&nbsp;"createdDate": "2019-05-06     <br />&nbsp;&nbsp;&nbsp;&nbsp;"createdByUserId": "28d1057c-…"  <br />&nbsp;&nbsp;}<br /> }|

2. With RMB method

| id  | jsonb | metadata |
| ------------- | -------------------------- | ------------- |
| a9b99f8a-… | { <br /> &nbsp;&nbsp;"id": "a9b99f8a-…", <br />  &nbsp;&nbsp;"accountingCode": "1234", <br />&nbsp;&nbsp;"amount": 23.45,  <br />&nbsp;&nbsp;"batchNumber": "101",  <br /> &nbsp;&nbsp;"disbursementNumber": "EFT5467",  <br />&nbsp;&nbsp;"disbursementDate": "2019-05-05",   <br />&nbsp;&nbsp;"disbursementAmount": 5,  <br />&nbsp;&nbsp;"exchangeRate": 0.89,  <br />&nbsp;&nbsp;"exportToAccounting": false,   <br />&nbsp;&nbsp;"invoiceCurrency": "USD",   <br />&nbsp;&nbsp;"invoiceId": "733cafd3-…",  <br />&nbsp;&nbsp;"status": "Awaiting payment",  <br />&nbsp;&nbsp;"systemCurrency": "EUR",  <br /> &nbsp;&nbsp;"type": "Payment",  <br />&nbsp;&nbsp;"voucherDate": "2019-05-06",  <br /> &nbsp;&nbsp;"voucherNumber": "1000",   <br />&nbsp;&nbsp;"metadata": {     <br />&nbsp;&nbsp;&nbsp;&nbsp;"createdDate": "2019-05-06     <br />&nbsp;&nbsp;&nbsp;&nbsp;"createdByUserId": "28d1057c-…"  <br />&nbsp;&nbsp;}<br /> }  | { <br /> &nbsp;&nbsp;"id": "0b44b189-...", <br />&nbsp;&nbsp;"recordId": "a9b99f8a-…", <br />&nbsp;&nbsp;"acquisitionsUnitId": "0ebb1f7d-…",   <br />&nbsp;&nbsp;"metadata": {     <br />&nbsp;&nbsp;&nbsp;&nbsp;"createdDate": "2019-05-06     <br />&nbsp;&nbsp;&nbsp;&nbsp;"createdByUserId": "28d1057c-…"  <br />&nbsp;&nbsp;}<br /> }|

Reply:
@pautau-s the story is clear that the RMB approach should be used here.  We discussed this as a team when grooming this story.  It would be helpful to implement one or two views this way for comparison purposes.  Creating views via custom sql scripts gives is more control but it also has it's drawbacks.  I suspect the RMB approach makes it easier to maintain and may also help when it comes to data migration scripts used to fix breaking schema changes upon upgrade.

Anyway, I was looking at the mod-users code it doesn't look too much more complicated, though DISTINCT ON isn't leveraged there.  See https://github.com/folio-org/mod-users/blob/8592e100b29e88fa8b52e5ae15ff884846c58323/src/main/java/org/folio/rest/impl/UsersAPI.java#L93 and https://github.com/folio-org/mod-users/blob/8592e100b29e88fa8b52e5ae15ff884846c58323/src/main/java/org/folio/rest/impl/UsersAPI.java#L132

Reply:
Just a thought... maybe keep this branch as-is and implement this the RMB way in a separate branch.  That would allow for an easy comparison of the two approaches (at the code level).  Such a comparison would probably be useful.